synapseclient/client.py

```diff
@@ -508,26 +508,26 @@ def login(
                 )
             )
 
-    def _get_config_section_dict(self, section_name):
+    def _get_config_section_dict(self, section_name: str) -> dict:
         config = self.getConfigFile(self.configPath)
         try:
             return dict(config.items(section_name))
         except configparser.NoSectionError:
             # section not present
             return {}
 
-    def _get_config_authentication(self):
+    def _get_config_authentication(self) -> str:
         return self._get_config_section_dict(
             config_file_constants.AUTHENTICATION_SECTION_NAME
         )
 
-    def _get_client_authenticated_s3_profile(self, endpoint, bucket):
+    def _get_client_authenticated_s3_profile(self, endpoint: str, bucket: str) -> dict:
```

Comment:
I'm not sure that this one is a `-> dict` in this case. I think it might be a `-> str`. My reasoning:
This is called in these spots:

1. https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/synapseclient/core/upload/upload_functions.py#L280
2. https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/synapseclient/client.py#L2401

It's then getting passed over to: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/synapseclient/core/remote_file_storage_wrappers.py#L54

And it looks like it's being used as a `str`

Reply:
Ah, yeah you're right. My mistake.

Reply:
Fixed in latest commit